src/storage/txn/latch.rs

```diff
@@ -19,7 +20,7 @@ const WAITING_LIST_SHRINK_SIZE: usize = 32;
 #[derive(Clone)]
 struct Latch {
     // store waiting commands
-    pub waiting: VecDeque<u64>,
+    pub waiting: VecDeque<Option<(usize, u64)>>,
```

Comment:
More memory?

Reply:
Just three times memory per Latch. But we could allocate less capacity of Latch. So it cost less memory in general.

Reply:
I think it's three times per task instead of per latch. It's hard to say whether the memory usage is reduced.

Reply:
No. In product environment, user usually use a much larger scheduler-concurrency to avoid high latch-wait latency. For example,  a bank may set scheduler-concurrency 102400000, which is five times of default value and cost more than 400 MB (pointer in VecDeque also cost memory) even if TiKV does not access any write request. 
According to my implementation, you could only need set scheduler-concurrency  512000,   which is 1/200 of 102400000 and cost no more than 2MB.
If there are 1000k keys ready to write into TiKV server, every key would cost 3 * 8B memory. All of they just cost 24MB memory, which is much less master.
So this PR would save much memory.